test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+		},
+	}
+	err = client.Create(context.TODO(), subscription)
+	if err != nil {
+		log.Error("Error create Operator Subscription in "+apicastNamespace+" namespace", err)
+		return err
+	}
+	return nil
+}
+
+// Create a self-managed APIcast
+func createSelfManagedApicast(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Create self-managed APIcast")
+	specConfigurationLoadMode := "boot"
+	//specImage := "image-registry.openshift-image-registry.svc:5000/selfmanaged-apicast/apicast-policy:example"
+	specReplicas := int64(1)
+	apicast := &appsv1alpha1.APIcast{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example-apicast",
+			Namespace: apicastNamespace,
+			Labels: map[string]string{
+				"operators.coreos.com/apicast-operator.selfmanaged-apicast": "",
+			},
+			Generation: 2,
+			Annotations: map[string]string{
+				"apicast.apps.3scale.net/operator-version": "0.5.0",
+			},
+			ManagedFields: []metav1.ManagedFieldsEntry{
+				{APIVersion: "apps.3scale.net/v1alpha1"},
+			},
+		},
+		Spec: appsv1alpha1.APIcastSpec{
+			ConfigurationLoadMode: &specConfigurationLoadMode,
+			//Image:                 &specImage,
+			Replicas: &specReplicas,
+			AdminPortalCredentialsRef: &corev1.LocalObjectReference{
+				Name: "adminportal-credentials",
+			},
+		},
+	}
+	err := client.Create(ctx, apicast)
+	if err != nil {
+		log.Error("Error create APIcast - example-apicast", err)
+		return err
+	}
+	return nil
+}
+
+// Create a route for the self-managed APIcast
+func createApicastRoute(ctx context.Context, client k8sclient.Client, threeScaleAdminPortal string) (string, error) {
+	log.Info("Create route for self-managed APIcast")
+	hostsub := strings.TrimPrefix(threeScaleAdminPortal, "3scale-admin")
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "apicast-example",
+			Namespace: apicastNamespace,
+			Labels: map[string]string{
+				"app":                  "apicast",
+				"threescale_component": "apicast",
+			},
+			Annotations: map[string]string{
+				"openshift.io/host.generated": "true",
+			},
+		},
+		Spec: routev1.RouteSpec{
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString("proxy"),
+			},
+			TLS: &routev1.TLSConfig{
+				Termination: "edge",
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: "apicast-example-apicast",
+			},
+			Host: "selfmanaged-staging" + hostsub,
+		},
+	}
+	err := client.Create(ctx, route)
+	if err != nil {
+		log.Error("Error create Apicast Route", err)
+		return "", err
+	}
+	routeHost := route.Spec.Host
+	log.Info("routeHost: " + routeHost)
+	return routeHost, nil
+}
+
+func customerLogin() error {
+	log.Info("Customer Login")
+	token, err := getCustomerAminPasswordTokenFromTerm("Enter Customer Admin user Token, and press Enter: ")
+	if err != nil {
+		log.Error("Error login customer, can't get token: ", err)
+		return err
+	}
+	command := "oc login --token=" + token
+	err = run_shell_command(command)
```

Comment:
A question here: is the purpose of running this command to verify that the user can login or is it to test other commands based on this user login?

Reply:
I think the main purpose of this is to test the whole procedure as a user that customer would use.